File: frontend/src/models/crmTask.ts
export type CRMTaskType = 'call' | 'email' | 'meeting' | 'follow_up' | 'demo' | 'proposal' | 'other';
export type CRMTaskPriority = 'low' | 'medium' | 'high' | 'urgent';
export type CRMTaskStatus = 'pending' | 'in_progress' | 'completed' | 'cancelled' | 'deferred';
export type CRMRelatedToType = 'lead' | 'deal' | 'company' | 'contact';

export interface CRMTask {
	public_id: string;
	title: string;
	description?: string;
	task_type: CRMTaskType;
	priority: CRMTaskPriority;
	status: CRMTaskStatus;
	assigned_to: number;
	created_by: number;
	related_to_type?: CRMRelatedToType;
	related_to_id?: number;
	due_date: string;
	completed_date?: string;
	is_reminder_sent: boolean;
	reminder_before_minutes: number;
	outcome?: Record<string, any>;
	attachments?: any[];
	custom_fields?: Record<string, any>;
	created_at: string;
	updated_at: string;
	assigned_user?: any;
	creator?: any;
}

export interface CRMTaskCreate extends Omit<CRMTask, 'public_id' | 'created_at' | 'updated_at' | 'assigned_user' | 'creator' | 'completed_date' | 'is_reminder_sent' | 'created_by'> { }
export interface CRMTaskUpdate extends Partial<CRMTaskCreate> {
	completed_date?: string;
	is_reminder_sent?: boolean;
}

export interface CRMTaskPaginatedResponse {
	items: CRMTask[];
	total: number;
}

File: backend/app/schemas/crm_task.py
"""CRM Task Schemas"""

from typing import Optional, Dict, Any, List
from uuid import UUID
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field
from app.models.crm_task import CRMTaskStatus, CRMTaskType, CRMTaskPriority, CRMRelatedToType


class CRMTaskBase(BaseModel):
    title: str = Field(..., max_length=255)
    description: Optional[str] = None
    task_type: CRMTaskType
    priority: CRMTaskPriority = CRMTaskPriority.MEDIUM
    status: CRMTaskStatus = CRMTaskStatus.PENDING
    assigned_to: int
    related_to_type: Optional[CRMRelatedToType] = None
    related_to_id: Optional[int] = None
    due_date: datetime
    reminder_before_minutes: int = 30
    custom_fields: Optional[Dict[str, Any]] = None


class CRMTaskCreate(CRMTaskBase):
    pass


class CRMTaskUpdate(BaseModel):
    title: Optional[str] = Field(None, max_length=255)
    description: Optional[str] = None
    task_type: Optional[CRMTaskType] = None
    priority: Optional[CRMTaskPriority] = None
    status: Optional[CRMTaskStatus] = None
    assigned_to: Optional[int] = None
    related_to_type: Optional[CRMRelatedToType] = None
    related_to_id: Optional[int] = None
    due_date: Optional[datetime] = None
    completed_date: Optional[datetime] = None
    is_reminder_sent: Optional[bool] = None
    reminder_before_minutes: Optional[int] = None
    outcome: Optional[Dict[str, Any]] = None
    attachments: Optional[List[Dict[str, Any]]] = None
    custom_fields: Optional[Dict[str, Any]] = None


class CRMTaskRead(CRMTaskBase):
    id: int
    public_id: UUID
    created_by: int
    completed_date: Optional[datetime] = None
    is_reminder_sent: bool
    outcome: Optional[Dict[str, Any]] = None
    attachments: Optional[List[Dict[str, Any]]] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CRMTaskListResponse(BaseModel):
    items: List[CRMTaskRead]
    total: int
